gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionFactory.java

```diff
@@ -617,9 +618,12 @@ private Schema buildRecordSchema(JsonSchema schema, WorkUnitState workUnit, Stri
         } catch (UnsupportedDateTypeException e) {
           throw new UnsupportedOperationException(e);
         }
-
-        Schema.Field fld = new Schema.Field(map.getColumnName(), fldSchema, map.getComment(),
-            map.isNullable() ? JsonNodeFactory.instance.nullNode() : null);
+        // TODO - Fix this. The "map" variable is of type JsonSchema (Gobblin internal
+        // implementation). For map.isNullable, AvroCompatibilityUtils.NULL_VALUE is passed. For
+        // non null, identify the exact change. As curently null is passed,
+        // JsonElementConversionFactoryTest is failing.
```

Comment:
> As curently null is passed, JsonElementConversionFactoryTest is failing.

When map.isNullable() is false, even the previous code was passing null, so there doesn't seem to be any change in that respect. So why does the test fail now, but didn't before?